server/src/CompilationDatabase.cpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * \file
+ *
+ * \brief Using libclang for reading the JSON compilation database.
+ *
+ * This file is distributed under the GNU General Public License. See
+ * COPYING for details.
+ */
+#include <iostream>
+
+#include <clang-c/Index.h>
+#include <clang-c/CXCompilationDatabase.h>
+#include <clang-c/CXString.h>
+
+#include "CompilationDatabase.h"
+
+static void getExactFlags(std::vector<std::vector<std::string>> &flags,
+                          CXCompilationDatabase &db,
+                          const std::string &fullFilename) {
+  unsigned ncmd;
+  unsigned i;
+  CXCompileCommands cmds;
+
+  cmds = clang_CompilationDatabase_getCompileCommands(db, fullFilename.c_str());
```

Comment:
In C++ I try to assign the variable directly, and not use the C method of first declaring all variable and then assign them.

```cpp
CXCompileCommands  cmds = clang_CompilationDatabase_getCompileCommands(db, fullFilename.c_str());
unsigned ncmd = clang_CompileCommands_getSize(cmds);

for (unsigned i = 0; i < ncmd; ++i) {
  ...
}
```